superset-frontend/src/datasource/ChangeDatasourceModal.tsx

```diff
@@ -20,25 +20,46 @@ import React, {
   FunctionComponent,
   useState,
   useRef,
-  useMemo,
   useEffect,
+  useCallback,
 } from 'react';
 import { Alert, FormControl, FormControlProps } from 'react-bootstrap';
-import { SupersetClient, t } from '@superset-ui/core';
+import { SupersetClient, t, styled } from '@superset-ui/core';
 import TableView from 'src/components/TableView';
-import Modal from 'src/common/components/Modal';
+import StyledModal from 'src/common/components/Modal';
+import Button from 'src/components/Button';
+import { useListViewResource } from 'src/views/CRUD/hooks';
+import Dataset from 'src/types/Dataset';
 import { getClientErrorObject } from '../utils/getClientErrorObject';
 import Loading from '../components/Loading';
 import withToasts from '../messageToasts/enhancers/withToasts';
 
+const CONFIRM_WARNING_MESSAGE = t(
+  'Warning! Changing the dataset may break the chart if the metadata does not exist in the target dataset',
```

Comment:
I'm not sure what this message means, can we clarify what metadata is necessary?